jobs/dap-collector-ppa-prod/dap_collector_ppa_prod/main.py

```diff
@@ -273,12 +296,18 @@ def check_time_precision(time_precision_minutes, end_collection_date):
     elif time_precision_minutes % MINUTES_IN_DAY != 0:
         # time precision is a day or longer, but is not a multiple of a day
         return f"Task has time precision that is not an even multiple of a day"
-    elif end_collection_date_seconds % (time_precision_minutes*60) != 0:
+    elif (end_collection_date_seconds + ad_start_offset) % (time_precision_minutes*60) != 0:
         # time precision is a multiple of day, but the end does not align with this task's buckets
         return f"{end_collection_date} does not align with task aggregation buckets"
 
     return None
 
+def find_ad_start_offset(task):
+    if "ad_start_date_iso" not in task:
```

Comment:
I think we can just name this `start_date` and have it set to a specific day (i.e. 2024-10-16).

We should also include an `end_date` and not collect if the current day is outside of the start-end range. This way we can stage tasks early without the collector collecting for the tasks.

I think we should also validate that these properties are present in tasks.json for all tasks before collecting rather than defaulting to 0 offset.

Reply:
ah in this case i guess we'll also need to check if the time_precision (btw why it's called time_precision while it seems to be the scraping interval?) value would result a full coverage between start_date and end_date, i think i'll do a check and throw a warning if not and continue -- a run with a partial last slot is better than nothing i think?